### ISSUE TITLE
[bug] N build_type -> 1 package cannot depends on package relying on build_type themselve 

### ISSUE THREAD

**Issue by Siron777** [conan-io/conan#7184]
Hi,

When a package C depends on a package S, and both are build in a _N build type-> 1 package_ mode:

- the _create_ command for the package C will not install the multiple build type necessary of S to compile C. It will only install the default build_type or the one specified with _-s build_type_. 
- And according to the question https://github.com/conan-io/conan/issues/5341, if the package S is not a _N build type-> 1 package_, it will also only download the package S with the default build_type or the one specified with _-s build_type_.

I am missing something or Conan is lacking a mechanism to specify which build type of S the package C will really need ?

Simon

**Comment by jgsogo** [comment]
Hi, @Siron777 . Yes, this is a limitation of "multi-config packages", all the packages they require have to be also "multi-config packages".

I would advise you not to use this kind of packages, it is much better to create different packages based on `build_type` if they are actually different binaries. Regular packages have many advantages: small packages, less time to download, less time to compile,... and `_multi` generators allow a developer to use a multi-config generator like VisualStudio or XCode even if different `build_type` result in different packages. 

In fact, we are thinking about deprecating the `cpp_info.<config>` syntax that these packages are using, so in fact, these packages would be deprecated themselves.

If this is something that affects you badly, please let us know, we are open to reconsidering these decisions, and right now we are defining the next Conan v2 release.

Thanks!

**Comment by Siron777** [comment]
Hi @jgsogo,

Well, I won't mind that much if those packages are deprecated. I am using them for convenience, as we are always using the debug and release build, but I am discovering that this convenience is not that big with the limitation describe here. The multi config package that we are not creating ourself come with their own cmake files (ex. Qt), for those ones we do not depend on the cmake multi generator.  My opinion is indeed that it is better to remove a feature than to keep it half supported. I would just suggest now to document that pending decision for future people considering using that feature.

Thank you,

Simon

**Comment by jgsogo** [comment]
You are right. Nevertheless, we cannot remove this functionality due to the stability commitment for versions 1.x, but this is something we should add to the [CONAN_V2_MODE](https://docs.conan.io/en/latest/reference/conan_v2_mode.html#conan-v2-mode) where we are starting to remove some functionality that we will deprecate in Conan v2.

**Comment by AbrilRBS** [comment]
Closing as Conan 1-only, which no further support is planned for.

Note that Conan 2 introduced some improvements over binary compatibility, and also allows to use the same build artifacts to package different package configurations
